### PR TITLE
add nmea gps

### PR DESCRIPTION
**Pull request by KmakD**
### Description

Moved from https://github.com/husarion/nmea-gps-docker/

### Modifications

-

## Summary by CodeRabbit

- **New Features**
	- Introduced a configuration file for NMEA GPS data handling, enhancing device management within the ROS ecosystem.
	- Added a launch argument for optional inclusion of the NMEA GPS driver in the localization process.
	- Created a new launch file for configuring the GPS device with customizable runtime parameters.
	- Enhanced API documentation with new GPS functionality, including parsing NMEA strings and publishing NavSat messages.

- **Chores**
	- Updated package dependencies to include additional libraries for improved functionality in navigation and data handling.
	- Improved documentation for configuration files and components related to GPS integration.

### CODE REVIEW

File: panther_localization/launch/nmea_navsat.launch.py [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+        default_value=EnvironmentVariable("ROBOT_NAMESPACE", default_value=""),
+        description="Namespace to all launched nodes and use namespace as tf_prefix. This aids in differentiating between multiple robots with the same devices.",
+    )
+
+    rename_params_file = ReplaceString(
+        source_file=params_file,
+        replacements={"<device_namespace>": device_namespace, "//": "/"},
+    )
+
+    nmea_driver = Node(
```
Comment (pawelirh) [inline panther_localization/launch/nmea_navsat.launch.py]: In this project, node object definitions in launch files include the `_node` suffix.

```suggestion
    nmea_driver_node = Node(
```

File: panther_localization/config/nmea_navsat_params.yaml
Comment (pawelirh) [inline panther_localization/config/nmea_navsat_params.yaml:1]: Please list this config in README

File: panther_localization/launch/nmea_navsat.launch.py
Comment (pawelirh) [inline panther_localization/launch/nmea_navsat.launch.py:1]: Please list this launch in README

File: panther_localization/launch/localization.launch.py
Comment (pawelirh) [inline panther_localization/launch/localization.launch.py:1]: Update in master `README.md` and `ROS_API.md` files is missing.

- new argument
- new nodes
- new topics

**Review by pawelirh — CHANGES_REQUESTED** [review]

File: ROS_API.md [hunk 2]
```diff
@@ -47,6 +47,7 @@ Below is information about the physical robot API. For the simulation, topics an
...
 | 🤖   | `hardware_controller`     | Plugin responsible for communicating with engine controllers via the CAN bus and providing E-Stop functionalities. <br/> *[panther_hardware_interfaces/PantherSystem](./panther_hardware_interfaces/src/panther_system/)*                                                                                                                                                                                                                                                                                                             |
+| 🤖   | `gps`                     | Node responsible parsing NMEA strings and publishing standard ROS NavSat message types. <br/> *[nmea_navsat_driver/nmea_navsat_driver](https://github.com/ros-drivers/nmea_navsat_driver/tree/ros2/src/libnmea_navsat_driver)*                                                                                                                                                                                                                                                                                                                                 |
```
Comment (pawelirh) [inline ROS_API.md]:
```suggestion
| 🤖   | `gps`                     | Node responsible for parsing NMEA strings and publishing standard ROS NavSat message types. <br/> *[nmea_navsat_driver/nmea_navsat_driver](https://github.com/ros-drivers/nmea_navsat_driver/tree/ros2/src/libnmea_navsat_driver)*                                                                                                                                                                                                                                                                                                                                 |
```

Comment (pawelirh) [inline ROS_API.md]: We should stick to the convention and change the name of this node to sth like `gps_driver` `nmea_gps_driver` etc...

Topics in current form are properly defined, so if node name is changes we would need to change remap rule as well to `gps/<topic_name>`

Comment (KmakD) [reply ROS_API.md]: I purposed changes in following commit

**Review by pawelirh — APPROVED** [review]

**Pull request merged**